src/audio/kpb.c

```diff
@@ -531,17 +539,105 @@ static void kpb_init_draining(struct comp_data *kpb, struct kpb_client *cli)
 	} else if (!is_sink_ready) {
 		trace_kpb_error("kpb_init_draining() error: "
 				"sink not ready for draining");
+
+	} else if (!kpb_have_enough_history_data(buff, history_depth)) {
+		trace_kpb_error("kpb_init_draining() error: "
+				"not enough data in history buffer");
 	} else {
+		/* draining accepted, find proper buffer to start reading */
+		while (buff->next && buff->next != first_buff) {
```

Comment:
@mrajwa if there is only buffer in the list, you will never check if it is in the FREE state?